crypto/http/http_client.c

```diff
@@ -1304,7 +1304,8 @@ BIO *OSSL_HTTP_get(const char *url, const char *proxy, const char *no_proxy,
         if (resp == NULL && redirection_url != NULL) {
             if (redirection_ok(++n_redirs, current_url, redirection_url)
                     && may_still_retry(max_time, &timeout)) {
-                (void)BIO_reset(bio);
+                if (bio != NULL)
```

Comment:
You will have to debug this since BIO_reset() calls BIO_ctrl() which does a NULL check

Reply:
or we are looking at different code :)